saleor/graphql/payment/mutations.py

```diff
@@ -105,6 +106,21 @@ def clean_payment_amount(cls, info, checkout_total, amount):
                 }
             )
 
+    @classmethod
+    def validate_gateway(cls, gateway_id, currency):
+        available_gateways = get_plugins_manager().list_payment_gateways(
+            currency=currency
+        )
+        if not any([gateway["id"] == gateway_id for gateway in available_gateways]):
```

Comment:
Not sure if it's needed anywhere else but this could be also a utility function in the payment module, e.g. `is_supported_currency(gateway, currency)`.